### ISSUE TITLE
Ctrl+Click on Vector2.move_toward() brings up documentation for @GDScript.move_toward()

### ISSUE THREAD

**Issue by Eerywax** [godotengine/godot#71136]
### Godot version

3.5.1 stable

### System information

Windows 10

### Issue description

I was trying to refine a function and ctrl+clicked move_toward() to make sure the change I wanted to make would work. I got very confused for a moment because it took me to @GDScript's function instead of Vector2's (they return different data types).

### Steps to reproduce

The whole script is below. It's attached to a Node2D that is childed to a projectile (Area2D), and is meant to handle specific movement behavior in the parent.

```
extends Node2D

func do_movement(delta: float, weapon_stats: Resource, target_pos: Vector2) -> void:
	get_parent().global_position = get_parent().global_position.move_toward(target_pos, delta * weapon_stats.speed)
```

### Minimal reproduction project

N/A

**Comment by AThousandShips** [comment]
Suspect this is because get_parent() is not guaranteed to return a Node2D and therefore `global_position` isn't guaranteed to be a Vector2

**Comment by Eerywax** [comment]
That makes sense. Should I just close this then?

**Comment by AThousandShips** [comment]
I think the fact that it brings up the `@GDScript` one should be investigated because it seems wrong

**Comment by akien-mga** [comment]
Yes it shouldn't fallback to `@GDScript` or `@GlobalScope` when using member/indexing syntax (`object.method`), as global methods are not available this way.

**Comment by kleonc** [comment]
> ### Steps to reproduce
> 
> The whole script is below. It's attached to a Node2D that is childed to a projectile (edit: also a Node2D), and is meant to handle specific movement behavior in the parent.
> 
> ```
> extends Node2D
> 
> func do_movement(delta: float, weapon_stats: Resource, target_pos: Vector2) -> void:
> 	get_parent().global_position = get_parent().global_position.move_toward(target_pos, delta * weapon_stats.speed)
> ```

For reference it works properly in v4.0.3.stable.official [5222a99f5]:
- With the scene opened (so the types in `get_parent().global_position` can be inferred) <kbd>Ctrl</kbd>-clicking `move_toward` brings `Vector2.move_toward` docs.
- With no scene opened / only the script opened the `move_toward` is not <kbd>Ctrl</kbd>-clickable.
